standalone/src/main/java/filodb/standalone/SimpleProfiler.java

```diff
@@ -0,0 +1,468 @@
+package filodb.standalone;
+
+import java.io.FileOutputStream;
+import java.io.InterruptedIOException;
+import java.io.IOException;
+import java.io.OutputStream;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+
+import java.lang.reflect.Method;
+
+import java.time.Instant;
+
+import java.nio.charset.StandardCharsets;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Simple profiler which samples threads and periodically logs a report to a file. When the
+ * process is cleanly shutdown, the profiler stops and reports what it has immediately. This
+ * makes it possible to use a very long report interval without data loss.
+ */
+public class SimpleProfiler {
+    /**
+     * Launches a global profiler, based on system properties.
+     *
+     * -Dfilodb.SimpleProfiler.sampleRateMillis=<n>
+     * -Dfilodb.SimpleProfiler.reportIntervalSeconds=<n>
+     * -Dfilodb.SimpleProfiler.topCount=<n>
+     * -Dfilodb.SimpleProfiler.outFile=<filename>
+     *
+     * In order for profiling to be enabled, all of the above properies must be set.
+     *
+     * @throws IOException if file cannot be opened
+     */
+    public static boolean launch() throws IOException {
+        String prefix = "filodb.SimpleProfiler.";
+
+        int sampleRateMillis = getIntProperty(prefix, "sampleRateMillis");
+        if (sampleRateMillis <= 0) {
+            return false;
+        }
+
+        int reportIntervalSeconds = getIntProperty(prefix, "reportIntervalSeconds");
+        if (reportIntervalSeconds <= 0) {
+            return false;
+        }
+
+        int topCount = getIntProperty(prefix, "topCount");
+        if (topCount <= 0) {
+            return false;
+        }
+
+        String outFile = System.getProperty(prefix + "outFile");
```

Comment:
Should we write to a tmp file and print out the path in logs?
Otherwise same file will be overwritten and this may not work in local environments with multiple servers.

Especially true if and when we use same server typesafe config for both servers.

Reply:
Temp file should work fine if the file isn't specified.